checkout-core/src/main/java/com/adyen/checkout/core/image/ImageLoader.kt

```diff
@@ -58,16 +69,41 @@ object DefaultImageLoader : ImageLoader {
 
                 val bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.size)
 
+                cache[url] = bitmap
+
                 withContext(Dispatchers.Main) {
                     onSuccess(bitmap)
                 }
             } else {
-                onError(HttpException(response.code, response.message, null))
+                withContext(Dispatchers.Main) {
+                    onError(HttpException(response.code, response.message, null))
+                }
             }
         } catch (e: CancellationException) {
             call.cancel()
         } catch (e: IOException) {
-            onError(e)
+            withContext(Dispatchers.Main) {
+                onError(e)
+            }
+        }
+    }
+
+    companion object {
+
+        private const val LOW_MEMORY_PERCENT = 0.15
+        private const val DEFAULT_MEMORY_PERCENT = 0.2
+        private const val DEFAULT_MEMORY_MEGABYTES = 256
+        private const val BYTE_CONVERSION = 1024
+
+        private fun calculateInMemoryCacheSize(context: Context): Int = try {
```

Comment:
Is there a reason for this method to be static?

Reply:
Not really, you think it's better to move it out?

Reply:
I read a bit about it [here](https://softwareengineering.stackexchange.com/questions/215826/make-methods-that-do-not-depend-on-instance-fields-static) and don't have a strong opinion about it. Some say keeping functions where you don't access the member fields static helps extracting them later on if needed (probably because you can see immediately). I think functionally there's not much distinction, so we can keep it as is.